### ISSUE TITLE
Syntax highlight codefences in markdown

### ISSUE THREAD

**Issue by tolmasky** [subethaedit/SubEthaEdit#38]
So on the face of it this seems like it should be a pretty trivial manual addition, just keying of the code-fence's tag and using that syntax mode for the body. However, while I plan on making a quick commit of this in the immediate, it would be nice to have something more automatic.

As an initial step, we could have a build script that just auto-generates these nested modes for every language and inserts them into the markdown mode, although this still won't support user's custom modes. This would also require other modes to specify their desired code-fence keys (for example, javascript supports probably "javascript", "js", "jsx", and "node").

Ideally there'd be an expressive way of iterating over all other modes at runtime such that markdown magically has knowledge of every other mode.

1. Part of umbrella bug: https://github.com/subethaedit/SubEthaEdit/issues/7
2. Possibly relevant to https://github.com/subethaedit/SubEthaEdit/issues/5 , although it would be nice to have something in there before a bigger change.

**Comment by tolmasky** [comment]
As I'm going through more of these, I'm trying to find a way to balance some SyntaxDefinition additions without making them be super-specific to the markdown mode. Basically, I want:

1. A way to map all other SyntaxDefinitions
2. Grab their filename recognitions to generate the fences.

I hate turning XML into an imperative thing, but "conceptually", I want something like this:

```XML
    <modes>
            <state id="CodeBlock-{mode.name}" usesymbolsfrommode="{mode.name}" useautocompletefrommode="{mode.name}" foldable="yes" scope="meta.block.{mode.name.toLowerCase()}">
                <begin><regex>^```(?:([^\.]*.)?({mode.fileextensions.join("|")}|{mode.filinames.join("|")|{mode.name}|{mode.aliases.join("|")})</regex></begin>
                <end><regex>```</regex></end>
                <import mode = "{mode.name}" />
            </state>
   </modes>
```
**Something** like that. I'm basing the codefence recognition on GitHub's, which is in turn based on the [rouge](https://github.com/jneen/rouge) library. If we look at just ruby as an example, we see why we need to support a whole bunch of different "types" of things ( https://github.com/jneen/rouge/blob/master/lib/rouge/lexers/ruby.rb#L11 ):

file extensions:

'*.rb',
'*.ruby',
'*.rbw',
'*.rake',
'*.gemspec',
'*.podspec',
 '*.ru',
'*.prawn',
'*.arb'

filenames:

'Rakefile',
'Guardfile',
'Gemfile',
'Capfile',
'Podfile',
'Vagrantfile',
'Berksfile',
'Dangerfile'

language names and aliases:

'ruby'
'rb' (handled by *.rb above already).

Anyways, just want to post current status. As I mentioned above, might be a good intermediate step to just have a build step inline all these modes and create the markdown mode. But eventually we want it to pick up on user's custom modes, and ideally do so live without restarting.

**Comment by monkeydom** [comment]
I'm closing this one with the merge of pull request #65 so it doesn't get lost in the Changelog. Please do file a fresh issue for further improvements. 

Fixed in d9d2e93cc842f3c654f3f918807f3375930491e0
